check-submodule.py

```diff
@@ -0,0 +1,316 @@
+    )
+
+
+# Run rust-code-analysis on the chosen repository to compute metrics
+def run_rca(repo_dir: pathlib.Path, output_dir: pathlib.Path) -> None:
+    run_subprocess(
+        "cargo",
+        "run",
+        "--package",
+        "rust-code-analysis-cli",
+        "--",
+        "--metrics",
+        "--output-format=json",
+        "--pr",
+        "-p",
+        repo_dir,
+        "-o",
+        output_dir,
+    )
+
+
+# Make the difference between the two json metric files
+def get_json_diff(
+    first_file: pathlib.Path, second_file: pathlib.Path
+) -> T.Tuple[T.Dict[str, T.Any], T.Dict[str, T.Any]]:
+    with open(first_file, "r") as input_file:
+        t1 = json.load(input_file)
+
+    with open(second_file, "r") as input_file:
+        t2 = json.load(input_file)
+
+    diff = deepdiff.DeepDiff(t1, t2, ignore_order=True)
+
+    return (t1, diff)
+
+
+# Save in a dictionary the filename and the list of code spans associated to
+# the differences
+def get_metrics_diff_span(
+    first_json: T.Dict[str, T.Any], diff: T.Dict[str, T.Any]
+) -> T.Dict[str, T.List[T.Tuple[int, int]]]:
+    # Search for this pattern in the differences object
+    prog = re.compile(r"\['spaces'\]\[\d+\]")
+
+    output = {"name": first_json["name"], "spaces_spans": []}
+
+    for value in diff["values_changed"]:
+        val = "".join(prog.findall(value))
+        # Subtracting one because files starts from 0
+        start_line = eval(f'first_json{val}["start_line"]') - 1
+        end_line = eval(f'first_json{val}["end_line"]')
+        output["spaces_spans"].append((start_line, end_line))
+
+    # Print the path of the repository file containing the differences
+    print(first_json["name"])
+
+    return output
+
+
+# Dump minimal tests code in an output file
+def dump_minimal_tests(
+    code_spans_object: T.Dict[str, T.List[T.Tuple[int, int]]],
+    new_filename: pathlib.Path,
+    compare_dir: pathlib.Path,
+) -> None:
+    # Remove duplicates from the list of spans
+    spans_list = list(dict.fromkeys(code_spans_object["spaces_spans"]))
+
+    # Get filename
+    filename = code_spans_object["name"]
+
+    # Read code spans from the input source code
+    with open(filename, "r") as input_file:
+        lines = input_file.readlines()
+
+    # Write spans to output file
+    output_path = compare_dir / new_filename.stem
+    with open(output_path, "w") as output_file:
+        for span in spans_list:
+            output_file.write("Minimal test:\n")
+            output_file.write("".join(lines[span[0] : span[1]]) + "\n")
+
+
+# Dump json file of differences
+def dump_json_file(
+    stdout: str, new_filename: pathlib.Path, compare_dir: pathlib.Path
+) -> None:
+    # Load the json file of differences as an object
+    diff_json = json.loads(stdout)
+
+    # Dump json file of differences
+    output_path = compare_dir / new_filename.name
+    with open(output_path, "w") as output_file:
+        json.dump(diff_json, output_file, indent=2)
```

Comment:
You're parsing stdout and then dumping it again to json, why not writing the file directly instead?